src/hotspot/share/gc/g1/g1Analytics.cpp

```diff
@@ -301,12 +316,20 @@ double G1Analytics::predict_cleanup_time_ms() const {
   return predict_zero_bounded(_concurrent_mark_cleanup_times_ms);
 }
 
-size_t G1Analytics::predict_rs_length() const {
-  return predict_size(_rs_length_seq) + predict_size(_rs_length_diff_seq);
+size_t G1Analytics::predict_rs_length(bool for_young_gc) const {
+  if (for_young_gc || !enough_samples_available(_mixed_rs_length_seq)) {
+    return predict_size(_young_rs_length_seq) + predict_size(_young_rs_length_diff_seq);
+  } else {
+    return predict_size(_mixed_rs_length_seq) + predict_size(_mixed_rs_length_diff_seq);
+  }
 }
 
-size_t G1Analytics::predict_pending_cards() const {
-  return predict_size(_pending_cards_seq);
+size_t G1Analytics::predict_pending_cards(bool for_young_gc) const {
+  if (for_young_gc || !enough_samples_available(_young_pending_cards_seq)) {
```

Comment:
Why `_young_pending_cards_seq`? I'd expect `_mixed_rs_length_seq` just to match its neighbor.

Reply:
Typo from breaking up the big patch I had. Good catch! The correct sequence is `_mixed_pending_cards_seq` though.